general-superstaq/general_superstaq/service.py

```diff
@@ -159,12 +161,12 @@ def get_targets(
 
     def submit_qubo(
         self,
-        qubo: dict[tuple[()] | tuple[str | int] | tuple[str | int, str | int], int | float],
+        qubo: Mapping[tuple[TQuboKey, ...], float],
         target: str,
         repetitions: int = 1000,
         method: str | None = None,
         max_solutions: int = 1000,
```

Comment:
The utility of `max_solutions` may need to be re-evaluated in light of [updates](https://github.com/Infleqtion/server-superstaq/blob/73e2fca00ddb8870f224d5014d1a2bfcddc982f9/superstaq/request_converter.py#L434-L437) (related to https://github.com/Infleqtion/client-superstaq/pull/914), but I think its fine as far as the scope of this PR is concerned